### PR TITLE
vkd3d: Implement ID3D12GraphicsCommandList9 and ID3D12Device10.

### PR DESCRIPTION
**Pull request by sherief**
I based this on https://github.com/sherief/vkd3d-proton/commit/9185edb42a3219113e86e3a48f84ce271ac65b17.

### CODE REVIEW

**Comment by sherief** [comment]
MIT or Public Domain license.

File: libs/vkd3d/command_list_profiled.h [hunk 1]
```diff
@@ -464,7 +464,27 @@ static void STDMETHODCALLTYPE d3d12_command_list_DispatchMesh_profiled(d3d12_com
...
+}
+
+static void STDMETHODCALLTYPE d3d12_command_list_RSSetDepthBias_profiled(d3d12_command_list_iface *iface, FLOAT DepthBias, FLOAT DepthBiasClamp, FLOAT SlopeScaledDepthBias)
+{
+    COMMAND_LIST_PROFILED_CALL(iface, RSSetDepthBias, DepthBias, DepthBiasClamp, SlopeScaledDepthBias);
+}
+
+static void STDMETHODCALLTYPE d3d12_command_list_IASetIndexBufferStripCutValue_profiled(d3d12_command_list_iface *iface, D3D12_INDEX_BUFFER_STRIP_CUT_VALUE IBStripCutValue)
+{
+    COMMAND_LIST_PROFILED_CALL(iface, IASetIndexBufferStripCutValue, NumBarrierGroups, IBStripCutValue);
```
Comment (HansKristian-Work) [inline libs/vkd3d/command_list_profiled.h:484]: Copy paste error?

File: include/vkd3d_d3d12.idl [hunk 2]
```diff
@@ -4211,6 +4245,50 @@ interface ID3D12Device9 : ID3D12Device8
...
+        const D3D12_RESOURCE_DESC1 *desc,
+        D3D12_BARRIER_LAYOUT initial_layout,
+        const D3D12_CLEAR_VALUE *optimized_clear_value,
+        ID3D12ProtectedResourceSession *protected_session,
+        UINT32 num_castable_formats,
+        DXGI_FORMAT *castable_formats,
+        REFIID iid,
+        void **resource);
+
+    HRESULT STDMETHODCALLTYPE CreatePlacedResource2(ID3D12Heap *heap,
```
Comment (HansKristian-Work) [inline include/vkd3d_d3d12.idl:4273]: Don't use STDMETHODCALLTYPE here, that is implied and is failing the build.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Smol nits

File: libs/vkd3d/vkd3d_private.h [hunk 3]
```diff
@@ -2229,7 +2229,7 @@ struct vkd3d_dynamic_state
 };
 
 /* ID3D12CommandList */
-typedef ID3D12GraphicsCommandList6 d3d12_command_list_iface;
+typedef ID3D12GraphicsCommandList9 d3d12_command_list_iface;
```
Comment (HansKristian-Work) [inline libs/vkd3d/vkd3d_private.h:2232]: d3d12_device_iface also needs to update to 10. I think that's what's causing the flurry of warnings.

Comment (sherief) [reply libs/vkd3d/vkd3d_private.h:2232]: Done.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
There is still 22 warnings: https://github.com/HansKristian-Work/vkd3d-proton/actions/runs/4174228407

File: libs/vkd3d/vkd3d_private.h (hunk 3 above)
Comment (HansKristian-Work) [reply libs/vkd3d/vkd3d_private.h:2232]: There are missing conversions in e.g. d3d12_device_query_interface.
ID3D12Device9_QueryInterface(), etc.

**Comment by sherief** [comment]
Seems clean now: https://github.com/HansKristian-Work/vkd3d-proton/actions/runs/4180600840

Sorry about all the friction, still getting familiar with vkd3d-proton.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**